src/schemes/boundary/system.jl

```diff
@@ -27,8 +28,7 @@ struct BoundarySPHSystem{BM, NDIMS, IC, CO, M, IM, CA} <: BoundarySystem{NDIMS}
             typeof(initial_condition), typeof(coordinates),
             typeof(movement), typeof(ismoving), typeof(cache)}(initial_condition,
                                                                coordinates, boundary_model,
-                                                               movement,
-                                                               ismoving, cache)
+                                                               movement, ismoving, cache)
```

Comment:
I can add the `buffer` in the constructor for consistency, but it is not necessary if the field `buffer` is declared as `Nothing`.

I would omit it in the constructor since it is just a dummy field for these systems.

Reply:
Thinking more about it, it might be better to add an extra `nothing`, since this only works, when `buffer` is the last field.

Reply:
I didn't know this was possible:
```julia
julia> struct B
           a::Float64
           b::Nothing

           function B(a)
               new(a)
           end
       end

julia> B(1.0)
B(1.0, nothing)
```
Is this documented somewhere?
I guess it's best to explicitly add `nothing` and not rely on this feature.